tests/ut/runtime/test_replay.py

```
@@ -0,0 +1,339 @@
+def test_replay_skips_invalidation_when_recompile_false(tmp_path: Path) -> None:
+    work_dir = _make_build_output(tmp_path)
+    bin_file = _touch(work_dir / "cache" / "incore_aiv_foo.bin")
+    so_file = _touch(work_dir / "kernels" / "aiv" / "foo.so")
+    with patch.object(replay_module, "execute_compiled"):
+        replay(work_dir, recompile=False)
+    assert bin_file.exists()
+    assert so_file.exists()
+
+
+def test_replay_missing_kernel_config_raises(tmp_path: Path) -> None:
+    with pytest.raises(FileNotFoundError, match="kernel_config.py"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)